### PR TITLE
Initial outliers by threshold plot.

### PR DESCRIPTION
**Pull request by snim2**
**WORK IN PROGRESS -- DO NOT MERGE**

To do:
- [x] Write window size into x-labels
- [x] More horizontal space between subplots
- [x] x-ticks should be integer formatted
- [x] Gridlines should be drawn on integer values
- [x] Add more headroom to the y-limits so that markers are not obscured by gridlines / x-axis
- [ ] Add 20% margin around x-axis so that markers are not obscured ...
- [x] Slightly smaller markers

### CODE REVIEW

**Comment by ltratt** [comment]
As a really shallow comment, I suspect that we should observe convention and put `main` as the last function.

**Comment by snim2** [comment]
needs squashing...

File: plot_outliers_by_threshold.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+        results = json.loads(file_.read())
+        return results
+    return None
+
+
+def create_cli_parser():
+    """Create a parser to deal with command line switches.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('json_files',
```
Comment (ltratt) [inline plot_outliers_by_threshold.py]: Arguments need compacting.

**Comment by ltratt** [comment]
OK, one comment to address, then you can squash and we can merge.

**Comment by snim2** [comment]
Ready to squash?

**Comment by ltratt** [comment]
Yep!

**Comment by ltratt** [comment]
Ready to merge?

**Comment by snim2** [comment]
Yup!

**Comment by snim2** [comment]
Thanks.

**Pull request merged**